docs/high-availability.md

```diff
@@ -0,0 +1,177 @@
+# High Availability Kubernetes Clusters
+
+## Introduction
+This document describes how to build a high-available Kubernetes cluster.  This is a fairly advanced topic, 
+Users who merely want to experiment with Kubernetes are encouraged to use a configurations that are simpler to set up such as
+the simple [Docker based single node cluster instructions](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/getting-started-guides/docker.md),
+or try [Google Container Engine](https://cloud.google.com/container-engine/) for hosted Kubernetes.
+
+## Overview
+Setting up a truly reliable, highly available distributed system requires a number of distinct steps, it is akin to
+wearing underwear, pants, a belt, suspenders, another pair of underwear, and another pair of pants.  We go into each
+of these steps in detail, but a summary is given here to help guide and orient the user.
+
+The steps involved are as follows:
+   * Setting up a redundant, reliable storage layer
+   * Installing a health-checking daemon runner
+   * Turning up a clustered instance of the etcd key/value store
+   * Starting replicated, load balanced Kubernetes API servers
+   * Setting up master-elected Kubernetes scheduler and controller-manager daemons
+
+Here's what the system should look like when it's finished:
+![High availability Kubernetes diagram](high-availability/ha.png)
+
+Ready? Let's get started.
+
+## Initial set-up.
+The remainder of this guide assumes that you are setting up a 3-node clustered master, where each machine is running some flavor of Linux.
+Examples in the guide are given for Debian distributions, but they should be easily adaptable to other distributions.
+
+The easiest way to implement an HA Kubernetes cluster, is to start with an existing single-master cluster.  See [https://get.k8s.io](https://get.k8s.io)
```

Comment:
don't need a comma after "cluster"
Also, make it clearer you actually want them to run the script at get.k8s.io. Suggested fix: say "Follow the instructions at ..." rather than "See ..."